packages/transaction-controller/src/TransactionController.ts

```diff
@@ -381,7 +386,9 @@ export class TransactionController extends BaseController<
     this.hub.emit(`unapprovedTransaction`, transactionMeta);
 
     return {
-      result: this.processApproval(transactionMeta),
+      result: this.processApproval(transactionMeta, {
+        requireApproval: Boolean(requireApproval),
```

Comment:
If we know `requireApproval` is a boolean, why convert it to one? Do we use `requireApproval === true` or `requireApproval === false` when we check it later on? If not, then it seems like we could safely say:

``` suggestion
        requireApproval,
```

Reply:
Since `addTransaction` is a public method we cannot foresee if it will be defined or not. Simply that conversion is only to satisfy TS(I didn't want `processApproval` to expect falsy statements). But for sure, this will not affect the flow in any way.